File: python/powx_n.py
class Solution:
    # @param x, a float
    # @param n, a integer
    # @return a float
    def pow(self, x, n):
        if n == 0:
            result = 1.0
        elif n < 0:
            result = 1.0/pow(x, -1*n)
        else:
            result = x * pow(x, n-1)
        
        return result

File: cpp/powx_n.cpp
class Solution {
public:
    double pow(double x, int n) {
        double result;
        if (n==0 || x == 1.0)
            result = 1.0;
        else if (x == -1.0)
        {
            if (n % 2 == 0)
                result = 1.0;
            else
                result = -1.0;
        }
        else if (n < 0)
            result = 1.0 / pow(x, -1*n);
        else
        {
            double tmp = pow(x, n/2);
            if (n%2 == 0)
                result = tmp *tmp;
            else
                result = x * tmp * tmp;
        }
        
        
        return result;
    }
};